src/pages/ImportData.js

```diff
@@ -154,8 +182,31 @@ const ImportData = () => {
                     borderTop: `${mapindex ? 1 : 0}px solid #AAA`
                   }
                 }, [
-                  icon('success-standard', { size: 18, style: { position: 'absolute', left: 0, color: colors.primary() } }),
-                  title
+                  !snapshotResponses ?
+                    div([
+                      icon('success-standard', { size: 18, style: { position: 'absolute', left: 0, color: colors.primary() } }),
+                      title
+                    ]) :
+                    Utils.switchCase(snapshotResponses[mapindex].status,
```

Comment:
This might be overkill but I just wanted to offer it anyway - not a blocker just a thought on DRYing this up a little since each option in the switch uses the same Fragment with icon and message template. So essentially making a function to return the filled out Fragment and then using the status to populate an object with the needed different things and passing that to the Fragment builder function

(this is not working code more just pseudo code)

```
> // A fragment builder  using the repeated style parts and taking as args the things that differ
>
>                       getMessageFragment( ({iconType,color, messageArray} = ...responseOptions )) {
>                         return h(Fragment, [
>                             icon(iconType, { size: 18, style: { position: 'absolute', left: 0, color } }),
>                             title,
>                             div({ style: { color, fontWeight: 'normal', fontSize: '0.625rem', marginTop: 5 } }, messageArray)
>                           ])
>                       }
>
>// the object of pieces specific to each response type
>                       const responseOptions = snapshotResponses[mapindex].status === 'fulfilled' ?
>                         {iconType: 'success-standard', color: colors.primary(), messageArray:  [strong(['Success: ']), 'Snapshot successfully imported']} : 
>                         {iconType: 'rejected', color: colors.danger(), messageArray:  [strong(['Error: ']),snapshotResponses[mapindex].message]
>                 } 
>
> // calling and returning the built fragment
>                     return getMessageFragment(responseOptions)`
```

Reply:
Made this update, please let me know what you think